### PR TITLE
SQL: fix for lost code in debug macros

### PR DESCRIPTION
**Pull request by kevgs**
Simple fix for https://github.com/natsys/mariadb_10.2/issues/17

### CODE REVIEW

File: sql/table.cc [hunk 1]
```diff
@@ -7057,18 +7057,21 @@ int TABLE::update_default_fields()
...
   DBUG_ENTER("vers_update_fields");
 
   DBUG_ASSERT(versioned());
-  DBUG_ASSERT(!vers_start_field()->set_time());
-  DBUG_ASSERT(!vers_end_field()->set_max_timestamp());
+  bool res= !vers_start_field()->set_time();
+  DBUG_ASSERT(res);
+  res= !vers_end_field()->set_max_timestamp();
+  DBUG_ASSERT(res);
+  (void)res;
```
Comment (midenok) [inline sql/table.cc:7069]: What goal aims this statement?

```
(void)res;
```

**Review by midenok — COMMENTED** [review]
Thank you for the fix! Please, clarify the question below.

File: sql/table.cc (hunk 1 above)
Comment (kevgs) [reply sql/table.cc:7069]: It silences compiler warning "unused variable".

Comment (kevgs) [reply sql/table.cc:7069]: I have no such warning with my gcc5.4 but other compilers could complain.

File: sql/table.cc [hunk 2]
```diff
@@ -7057,18 +7057,21 @@ int TABLE::update_default_fields()
...
-bool TABLE::vers_update_fields()
+void TABLE::vers_update_fields()
 {
   DBUG_ENTER("vers_update_fields");
 
   DBUG_ASSERT(versioned());
-  DBUG_ASSERT(!vers_start_field()->set_time());
-  DBUG_ASSERT(!vers_end_field()->set_max_timestamp());
+  bool res= !vers_start_field()->set_time();
+  DBUG_ASSERT(res);
```
Comment (avodaniel) [inline sql/table.cc:7066]: I also use something like:

```
if (vers_start_field()->set_time())
     DBUG_ASSERT(0);
```

It doesn't need additional variable.

Comment (midenok) [reply sql/table.cc:7066]: Good point, but probably -O will eliminate it anyway.

**Pull request merged**